core/chains/evm/gas/fee_history_estimator.go

```diff
@@ -0,0 +1,414 @@
+	maxFeePerGas := baseFee.AddPercentage(BaseFeeBufferPercentage).Add(maxPriorityFeePerGas)
+
+	promFeeHistoryEstimatorBaseFee.WithLabelValues(u.chainID.String()).Set(float64(baseFee.Int64()))
+	promFeeHistoryEstimatorMaxPriorityFeePerGas.WithLabelValues(u.chainID.String()).Set(float64(maxPriorityFeePerGas.Int64()))
+	promFeeHistoryEstimatorMaxFeePerGas.WithLabelValues(u.chainID.String()).Set(float64(maxFeePerGas.Int64()))
+
+	u.logger.Debugf("Fetched new dynamic prices, nextBlock#: %v - oldestBlock#: %v - maxFeePerGas: %v - maxPriorityFeePerGas: %v - maxPriorityFeeThreshold: %v",
+		nextBlock, feeHistory.OldestBlock, maxFeePerGas, maxPriorityFeePerGas, priorityFeeThresholdWei)
+
+	u.dynamicPriceMu.Lock()
+	defer u.dynamicPriceMu.Unlock()
+	u.dynamicPrice.FeeCap = maxFeePerGas
+	u.dynamicPrice.TipCap = maxPriorityFeePerGas
+	return u.dynamicPrice, nil
+}
+
+func (u *FeeHistoryEstimator) getDynamicPrice() (fee DynamicFee, err error) {
+	u.dynamicPriceMu.RLock()
+	defer u.dynamicPriceMu.RUnlock()
+	if u.dynamicPrice.FeeCap == nil || u.dynamicPrice.TipCap == nil {
+		return fee, fmt.Errorf("dynamic price not set")
+	}
+	return u.dynamicPrice, nil
+}
+
+// BumpLegacyGas provides a bumped gas price value by bumping the previous one by BumpPercent.
+// If the original value is higher than the max price it returns an error as there is no room for bumping.
+// It aggregates the market, bumped, and max gas price to provide a correct value.
+func (u *FeeHistoryEstimator) BumpLegacyGas(ctx context.Context, originalGasPrice *assets.Wei, gasLimit uint64, maxPrice *assets.Wei, _ []EvmPriorAttempt) (*assets.Wei, uint64, error) {
+	// Sanitize original fee input
+	if originalGasPrice == nil || originalGasPrice.Cmp(maxPrice) >= 0 {
+		return nil, 0, fmt.Errorf("%w: error while retrieving original gas price: originalGasPrice: %s. Maximum price configured: %s",
+			commonfee.ErrBump, originalGasPrice, maxPrice)
+	}
+
+	currentGasPrice, err := u.getGasPrice()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	bumpedGasPrice := originalGasPrice.AddPercentage(u.config.BumpPercent)
+	bumpedGasPrice, err = LimitBumpedFee(originalGasPrice, currentGasPrice, bumpedGasPrice, maxPrice)
+	if err != nil {
+		return nil, 0, fmt.Errorf("gas price error: %s", err.Error())
+	}
+
+	u.logger.Debugw("bumped gas price", "originalGasPrice", originalGasPrice, "bumpedGasPrice", bumpedGasPrice)
+
+	return bumpedGasPrice, gasLimit, nil
+}
+
+// BumpDynamicFee provides a bumped dynamic fee by bumping the previous one by BumpPercent.
+// If the original values are higher than the max price it returns an error as there is no room for bumping. If maxPriorityFeePerGas is bumped
+// above the priority fee threshold then there is a good chance there is a connectivity issue and we shouldn't bump.
+// Both maxFeePerGas as well as maxPriorityFerPergas need to be bumped otherwise the RPC won't accept the transaction and throw an error.
```

Comment:
nit: typo
```suggestion
// Both maxFeePerGas as well as maxPriorityFeePergas need to be bumped otherwise the RPC won't accept the transaction and throw an error.
```